llama_stack/providers/utils/kvstore/redis/redis.py

```diff
@@ -48,5 +48,27 @@ async def delete(self, key: str) -> None:
     async def range(self, start_key: str, end_key: str) -> List[str]:
         start_key = self._namespaced_key(start_key)
         end_key = self._namespaced_key(end_key)
+        cursor = 0
+        pattern = start_key + "*"  # Match all keys starting with start_key prefix
+        matching_keys = []
+        while True:
+            cursor, keys = await self.redis.scan(cursor, match=pattern, count=1000)
 
-        return await self.redis.zrangebylex(start_key, end_key)
+            for key in keys:
+                key_str = key.decode("utf-8") if isinstance(key, bytes) else key
```

Comment:
why is the decode necessary? we already using a str as input when setting the key.

Reply:
So when we fetch the keys using .scan() method, the list of keys fetched is of instance bytes. I tested it on my local machine as well and I am getting the list of keys where each key is of type byte.
@dineshyv